file_writer_control/JobStatus.py

```diff
@@ -0,0 +1,101 @@
+from enum import Enum, auto
+from datetime import datetime
+
+
+class JobState(Enum):
+    """
+    The state of a job.
+    """
+    NO_JOB = auto()
+    WAITING = auto()
+    WRITING = auto()
+    TIMEOUT = auto()
+    ERROR = auto()
+    DONE = auto()
+    UNAVAILABLE = auto()
+
+
+class JobStatus:
+    """
+    Contains general information about the (execution) of a job.
+    """
+    def __init__(self, job_id: str):
+        self._job_id = job_id
+        self._service_id = ""
+        self._last_update = datetime.now()
+        self._state = JobState.WAITING
+        self._message = ""
+
+    def update_status(self, new_status):
+        """
+        Updates the status/state of a this instance of the JobStatus class, using another instance.
+        .. note:: The job identifier of this instance and the other must be identical.
+        :param new_status: The other instance of the JobStatus class.
+        """
+        if new_status.job_id != self.job_id:
+            raise RuntimeError(
+                "Job id of status update is not correct ({} vs {})".format(
+                    self.job_id, new_status.job_id
+                )
+            )
+        self._state = new_status.state
+        if new_status.message:
+            self._message = new_status.message
+        self._service_id = new_status.service_id
+        self._last_update = new_status.last_update
+
+    @property
+    def job_id(self) -> str:
```

Comment:
While this function has an explicit return type others have not. May be it is better to have a coherent codebase. Either we should have a typed code all together or not.

Reply:
I did not care to set the return type of functions that did not return anything. Nevertheless, the return type of those functions have now been set to `None`.

Reply:
Oh sorry ı was not entirely clear. Sure, if it obviously return None, we do not need it. But for example in update_status method the type of the new_status is still missing. That was what I meant, if we are going for typed, everything should be typed :)

Reply:
Ah, the Python interpreter does not accept type names of the enclosing class. However, it does accept strings it seems so I have used that now.